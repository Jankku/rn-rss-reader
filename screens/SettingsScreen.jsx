import ThemeSection from '../components/Settings/ThemeSection';
import AboutSection from '../components/Settings/AboutSection';

function SettingsScreen() {
  return (
    <>
      <ThemeSection />
      <AboutSection />
    </>
  );
}

export default SettingsScreen;
