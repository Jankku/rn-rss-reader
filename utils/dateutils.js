import { DateTime } from 'luxon';

/**
 * For articles released today return time and date otherwise.
 * @param {string} pubDate - RFC2822 date
 * @returns {string} Formatted date
 */
export function RFC2822ToTimeOrDate(pubDate) {
  const dateObject = DateTime.fromRFC2822(pubDate);
  const formattedDate =
    dateObject.toRelativeCalendar() === 'today'
      ? dateObject.toLocaleString(DateTime.TIME_24_SIMPLE)
      : dateObject.toLocaleString(DateTime.DATE_SHORT);
  return formattedDate;
}

/**
 * @param {string} pubDate - RFC2822 date
 * @returns {string} Formatted date in DateTime format
 */
export function RFC2822ToDateTime(pubDate) {
  const dateObject = DateTime.fromRFC2822(pubDate);
  const formattedDate = dateObject.toLocaleString(DateTime.DATETIME_SHORT);
  return formattedDate;
}
