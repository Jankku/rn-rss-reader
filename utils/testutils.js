import { render } from '@testing-library/react-native';
import { ThemeContextProvider } from '../context/ThemeContext';
import { LocationContextProvider } from '../context/LocationContext';
import { RegionContextProvider } from '../context/RegionContext';

const AllProviders = ({ children }) => {
  return (
    <LocationContextProvider>
      <RegionContextProvider>
        <ThemeContextProvider>{children}</ThemeContextProvider>
      </RegionContextProvider>
    </LocationContextProvider>
  );
};

const customRender = (ui, options) => render(ui, { wrapper: AllProviders, ...options });

// re-export everything
export * from '@testing-library/react-native';

// override render method
export { customRender as render };
