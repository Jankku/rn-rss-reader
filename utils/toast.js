import Toast from 'react-native-root-toast';

const showToast = (message) =>
  Toast.show(message, {
    position: 100,
    opacity: 0.9,
    shadow: false,
  });

export default showToast;
