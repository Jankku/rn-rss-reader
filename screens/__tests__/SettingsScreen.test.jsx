import SettingsScreen from '../SettingsScreen';
import { render } from '../../utils/testutils';

describe('Test SettingsScreen', () => {
  test('Should match snapshot', () => {
    const { toJSON } = render(<SettingsScreen />);
    expect(toJSON()).toMatchSnapshot();
  });
});
