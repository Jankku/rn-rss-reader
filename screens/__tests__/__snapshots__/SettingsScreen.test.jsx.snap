// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SettingsScreen Should match snapshot 1`] = `
Array [
  <Text
    style={
      Object {
        "color": "#101010",
        "fontSize": 22,
        "fontWeight": "700",
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    Theme
  </Text>,
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#f7f7f7",
        "padding": 16,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text />
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "flexGrow": 1,
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#101010",
              "fontSize": 16,
              "fontWeight": "700",
            }
          }
        >
          Dark mode
        </Text>
        <RCTSwitch
          accessibilityRole="switch"
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": 31,
              "width": 51,
            }
          }
          thumbTintColor="#0e65cc"
          value={false}
        />
      </View>
    </View>
  </View>,
  <RCTScrollView
    data={
      Array [
        Object {
          "data": Array [
            Object {
              "icon": "logo-android",
              "title": "mock",
              "value": "mock",
            },
            Object {
              "icon": "person",
              "link": "https://github.com/Jankku/",
              "title": "Developer",
              "value": "Jankku",
            },
            Object {
              "icon": "logo-github",
              "link": "https://github.com/Jankku/rn-rss-reader",
              "title": "GitHub",
              "value": "https://github.com/Jankku/rn-rss-reader",
            },
          ],
          "title": "About",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={
      Array [
        0,
      ]
    }
    updateCellsBatchingPeriod={50}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          style={
            Object {
              "color": "#101010",
              "fontSize": 22,
              "fontWeight": "700",
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            }
          }
        >
          About
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#f7f7f7",
                "padding": 16,
              }
            }
          >
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text />
              <View>
                <Text
                  style={
                    Object {
                      "color": "#101010",
                      "fontSize": 16,
                      "fontWeight": "700",
                    }
                  }
                >
                  mock
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#101010",
                    }
                  }
                >
                  mock
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "#dedede",
                "height": 1,
              }
            }
          />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#f7f7f7",
                "padding": 16,
              }
            }
          >
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text />
              <View>
                <Text
                  style={
                    Object {
                      "color": "#101010",
                      "fontSize": 16,
                      "fontWeight": "700",
                    }
                  }
                >
                  Developer
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#101010",
                    }
                  }
                >
                  Jankku
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "#dedede",
                "height": 1,
              }
            }
          />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#f7f7f7",
              "padding": 16,
            }
          }
        >
          <View
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text />
            <View>
              <Text
                style={
                  Object {
                    "color": "#101010",
                    "fontSize": 16,
                    "fontWeight": "700",
                  }
                }
              >
                GitHub
              </Text>
              <Text
                style={
                  Object {
                    "color": "#101010",
                  }
                }
              >
                https://github.com/Jankku/rn-rss-reader
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      />
    </View>
  </RCTScrollView>,
]
`;
